web/cobrands/fixmystreet/staff.js

```diff
@@ -342,3 +342,68 @@ $.extend(fixmystreet.set_up, {
     });
   }
 });
+
+fixmystreet.maps = fixmystreet.maps || {};
+
+$.extend(fixmystreet.maps, {
+  shortlist_multiple: function(ids, token, count) {
+    var retryCount = (typeof count !== 'undefined') ?  count : 0;
+    $.post("/my/planned/change_multiple", { ids: ids, token: token })
+    .done(function() {
+      var $itemList = $('.item-list'),
+          items = [];
+
+      for (var i = 0; i < ids.length; i++) {
+        var problemId = ids[i],
+            $item = $itemList.find('#report-'+ problemId),
+            $form = $item.find('form'),
+            $submit = $form.find("input[type='submit']" );
+
+        $item.attr('data-lastupdate', new Date().toISOString());
+
+        toggle_shortlist($submit, 'remove', problemId);
+
+        items.push({
+          'url': '/report/' + $item.data('report-id'),
+          'lastupdate': $item.data('lastupdate')
+        });
+      }
+      $(document).trigger('shortlist-all', { items: items});
+    })
+    .fail(function(response) {
+      if (response.status == 400 && retryCount < 4) {
+        // If the response is 400, then get a new CSRF token and retry
+        var csrf = response.responseText.match(/content="([^"]*)" name="csrf-token"/)[1];
+        fixmystreet.maps.shortlist_multiple(ids, csrf, retryCount + 1);
+      } else {
+        alert("We appear to be having problems. Please try again later.");
+      }
+    });
+  },
+
+  show_shortlist_control: function() {
+    var $shortlistButton = $('#fms_shortlist_all');
+    if ($shortlistButton === undefined || fixmystreet.page != "reports" ) {
+      return;
+    }
+
+    if (fixmystreet.map.getZoom() >= 14) {
+      $shortlistButton.removeClass('hidden');
+      $shortlistButton.on('click', function() {
```

Comment:
This will readd this listener every time the zoom goes above 14. It only needs to be added once on setup (or lazily on first time zoom hits 14), I think?